source/application/CLS/CLSAppController.cpp

```diff
@@ -60,6 +61,14 @@ bool CLSAppController::startup()
 	return false;
 }
 
+void CLSAppController::shutdown()
+{
+	if (userConfiguration_)
```

Comment:
We could implement a user configuration setter for the app controller at the CLS-level. Each beamline could set their particular configuration that way.